### DISCUSSION TITLE
errors with

### DISCUSSION THREAD

**Discussion by CodeWithJoe2020** [wevm/wagmi#3058] [General]
I keep getting this error with below code

An error occurred preparing the transaction: Execution reverted for an unknown reason. Raw Call Arguments: from: 0x293Bcb20584a2e0E6d1b559E25783311ec29121C to: 0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3 value: 0.01 SEP data: 0xb91038c7000000000000000000000000293bcb20584a2e0e6d1b559e25783311ec29121c Contract Call: address: 0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3 function: participate(address _referrer) args: (0x293Bcb20584a2e0E6d1b559E25783311ec29121C) sender: 0x293Bcb20584a2e0E6d1b559E25783311ec29121C Docs: https://viem.sh/docs/contract/simulateContract.html Details: execution reverted Version: viem@0.3.39

```

function Write() {
  const { config, error } = usePrepareContractWrite({
    address: '0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3',
    abi: lottoABI, // Use the imported ABI JSON file
    functionName: 'participate',
    value:parseEther('0.01'),
    args:['0x293Bcb20584a2e0E6d1b559E25783311ec29121C']
  })
  const { data, isLoading, isSuccess, write } = useContractWrite(config)
  console.log(data)

  return (
    <>
      <button disabled={!write || isLoading} onClick={() => write?.()}>
        {isLoading ? 'Loading...' : isSuccess ? 'Success!' : 'Bought'}
      </button>
      {error && (
        <div>An error occurred preparing the transaction: {error.message}</div>
      )}
    </>
  );
}
export default Write;
```

**Comment by GAEAlimited** [comment]
The error message you provided appears to be related to a smart contract transaction on the Viem platform. The error message indicates that the transaction execution reverted for an unknown reason, and it includes some details about the transaction, such as the sender's address, the recipient's address, the transaction value, and contract call details.

Here's a breakdown of the information in the error message:

from: This is the Ethereum address that initiated the transaction, which is 0x293Bcb20584a2e0E6d1b559E25783311ec29121C in this case.

to: This is the recipient's Ethereum address, which is 0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3.

value: This is the amount of cryptocurrency (in this case, it's likely in Ether) sent with the transaction, which is 0.01.

SEP data: This appears to be additional data or parameters sent with the transaction, possibly related to the smart contract call.

Contract Call: This section provides information about the contract call that was made as part of the transaction. It specifies the contract address (0x3262c6f03F4003F1b4D6162F40bae8A71a101ef3), the function that was called (participate(address _referrer)), and the arguments passed to the function (the address 0x293Bcb20584a2e0E6d1b559E25783311ec29121C as the _referrer argument).

sender: This is the Ethereum address of the sender, which matches the from address.

Docs: This provides a link to documentation related to the contract or transaction. In this case, it points to the Viem documentation for simulating contracts.

Details: This section indicates that the execution of the transaction reverted, but it doesn't specify the exact reason for the revert.

Version: This specifies the version of the Viem platform being used, which is viem@0.3.39.

To diagnose the exact cause of the error, you may need to refer to the Viem documentation or reach out to their support/community for assistance. The error message suggests checking the provided documentation link for more information on simulating contracts, which could help you understand the issue in more detail and potentially resolve it.

**Comment by CodeWithJoe2020** [comment]
viem and wagmi dont have any community or support
